example/android/app/build.gradle

```diff
@@ -21,27 +22,28 @@ if (flutterVersionName == null) {
     flutterVersionName = '1.0'
 }
 
-apply plugin: 'com.android.application'
-apply plugin: 'kotlin-android'
-apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
-
 android {
-    compileSdkVersion 30
+    compileSdkVersion flutter.compileSdkVersion
+    ndkVersion flutter.ndkVersion
 
     compileOptions {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
 
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
+
     sourceSets {
         main.java.srcDirs += 'src/main/kotlin'
     }
 
     defaultConfig {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
         applicationId "com.example.background_location_example"
-        minSdkVersion 18
-        targetSdkVersion 30
+        minSdkVersion flutter.minSdkVersion
+        targetSdkVersion flutter.targetSdkVersion
```

Comment:
maybe im still a little new to flutter, but im not seeing where these variables (`flutter.minSdkVersion` .etc) are being set - could you point me to where these are set?

Reply:
These would come based on the dev installed flutter versions rather than being "hard-coded" they are older flutter variables that were initially not updated by flutter themselves and where it is recommended for production applications to use manually set values, however for example applications making use of them is better as they will not require repeated updating and based on the supported/installed flutter version always be compatible.